payments/tests/test_payments_models.py

```diff
@@ -886,3 +887,33 @@ def test_order_set_status_no_application(berth):
     order.set_status(OrderStatus.PAID)
     assert order.status == OrderStatus.PAID
     assert order.lease.status == LeaseStatus.PAID
+
+
+@pytest.mark.parametrize(
+    "order",
+    ["berth_order", "winter_storage_order", "additional_product_order"],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "from_status,to_status",
+    [
+        (OrderStatus.WAITING, OrderStatus.PAID),
+        (OrderStatus.WAITING, OrderStatus.REJECTED),
+        (OrderStatus.PAID, OrderStatus.CANCELLED),
+    ],
+)
+@freeze_time("2020-01-01T08:00:00Z")
+def test_order_paid_and_created(order, from_status, to_status):
+    order.status = from_status
+    order.save()
+
+    order.set_status(to_status)
+    assert order.status == to_status
+
+    order = Order.objects.get(id=order.id)
+    if to_status == OrderStatus.PAID:
+        assert order.paid_at == now()
```

Comment:
Maybe add a assertion that order.paid_at/rejected_at/cancelled_at == None on a order without any matching OrderLogEntries?

```suggestion
        assert order.paid_at == now()
        assert order.rejected_at == None
        assert order.cancelled_at == None        
```

Reply:
Updated